packages/eslint-plugin/src/rules/prefer-optional-chain-utils/analyzeChain.ts

```diff
@@ -368,15 +404,22 @@ function getFixer(
     newCode = `!${newCode}`;
   }
 
-  const fix: ReportFixFunction = fixer =>
-    fixer.replaceTextRange(
-      [chain[0].node.range[0], lastOperand.node.range[1]],
-      newCode,
-    );
+  const reportRange = getReportRange(chain, node.range, sourceCode);
 
-  return useSuggestionFixer
-    ? { suggest: [{ fix, messageId: 'optionalChainSuggest' }] }
-    : { fix };
+  const fix: ReportFixFunction = fixer =>
+    fixer.replaceTextRange(reportRange, newCode);
+
+  return {
+    messageId: 'preferOptionalChain',
+    loc: {
+      start: sourceCode.getLocFromIndex(reportRange[0]),
+      end: sourceCode.getLocFromIndex(reportRange[1]),
+    },
+    suggest: useSuggestionFixer
+      ? [{ fix, messageId: 'optionalChainSuggest' }]
+      : undefined,
+    fix: !useSuggestionFixer ? fix : undefined,
```

Comment:
Consider (re)using a little helper like so:
https://github.com/typescript-eslint/typescript-eslint/blob/88a8673b185a9ad9d4d7f31059da806f5d9a7449/packages/eslint-plugin/src/rules/return-await.ts#L345-L348

Reply:
@kirkwaiblinger  Cool! Is it necessary to add fixOrSuggest to `packages/utils` to make it common (although I think it would be better to refactor it as a separate issue)?

Reply:
Nah I think it could just go in eslint-pluin/src/utils, right? If it's anything more complicated than that (i.e. involving cross-package boundaries) feel free to just copypaste and file followup to find a home for the utility

Reply:
I tried to move the helper logic to utils, but `require-await` causes a lint in the [eslint-plugin/require-meta-has-suggestions](https://github.com/eslint-community/eslint-plugin-eslint-plugin/blob/08a8a098d14eed3965563bd57fe7085dd62da00d/docs/rules/require-meta-has-suggestions.md) rule. Perhaps it would be nice to add an option to that rule or turn off the rule (that recognizes certain functions as suggestions), what do you think?

Reply:
oh, bleh. I defer to you 🤷 👍

Reply:
@kirkwaiblinger I fixed! 5c57531